templates/ortools/main.py

```diff
@@ -0,0 +1,101 @@
+"""
+Template for working with Google OR-Tools.
+"""
+
+import json
+import sys
+import time
+from typing import Any, Dict
+
+from ortools.linear_solver import pywraplp
+
+
+def main():
+    """Entry point for the template."""
+    input_data = read_input()
+    solution = solve(input_data)
+
+    # Writes to stdout.
+    print(json.dumps(solution, indent=2))
+
+
+def solve(input_data: Dict[str, Any]) -> Dict[str, Any]:
+    """Solves the given problem and returns the solution."""
+
+    # Creates the solver.
+    provider = "SCIP"
+    solver = pywraplp.Solver.CreateSolver(provider)
+
+    # Initializes the linear sums.
+    weights = 0.0
+    values = 0.0
+
+    # Creates the decision variables and adds them to the linear sums.
+    items = []
+    for item in input_data["items"]:
+        item_variable = solver.IntVar(0, 1, item["item_id"])
+        items.append({"item": item, "variable": item_variable})
+        weights += item_variable * item["weight"]
+        values += item_variable * item["value"]
+
+    # This constraint ensures the weight capacity of the knapsack will not be
+    # exceeded.
+    solver.Add(weights <= input_data["weight_capacity"])
+
+    # Sets the objective function: maximize the value of the chosen items.
+    solver.Maximize(values)
+
+    # Solves the problem.
+    start_time = time.time()
+    status = solver.Solve()
+    end_time = time.time()
+
+    # Determines which items were chosen.
+    chosen_items = []
+    for item in items:
+        if item["variable"].solution_value() == 1.0:
```

Comment:
Do we need to take into account numerical problems. E.g. `0.9999` as the solution value? We could also do `item["variable"].solution_value() > 0.9` or something

Reply:
Yep.